### PR TITLE
chore: enable husky pre-commit tasks

### PR DESCRIPTION
**Pull request by maryamtahhan**
### What does this PR do?
First in a series of changes to enable Husky, conventional commits and lint staged for the project

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/1633

### How to test this PR?

```
touch test
git add test
git commit -m "a commit message that should fail"
# OR
git commit -m "chore: some test implementation"
```

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
suggestion: here is a PR in another repository

https://github.com/containers/podman-desktop-extension-bootc/commit/abf874ec5c9f6801247a5f6e346a4431ae4608b7

**Comment by maryamtahhan** [comment]
> suggestion: here is a PR in another repository
> 
> [containers/podman-desktop-extension-bootc@abf874e](https://github.com/containers/podman-desktop-extension-bootc/commit/abf874ec5c9f6801247a5f6e346a4431ae4608b7)

As in just pull that change in? In all honesty I'm a husky newb - I did try something like this based on the `podman-desktop` commits and it wouldn't work for me out of the box :(... After following the documentation for the different "plugins" this is what ended up working in my environment. But I'm happy to close this PR if it's not what you are looking for. Hopefully someone else in the community can pick it up. I can still make the other contributions based on what the pre-commit hooks flagged yesterday.

**Comment by benoitf** [comment]
copy the commit-msg, pre-commit from the linked PR + introduce `lint-staged` in package.json + the devDependencies and then I think it'll work

**Comment by maryamtahhan** [comment]
> copy the commit-msg, pre-commit from the linked PR + introduce `lint-staged` in package.json + the devDependencies and then I think it'll work

Ok - I will give it another go

**Review by jeffmaury — APPROVED** [review]
LGTM and thank you for you patience @maryamtahhan

File: package.json [hunk 1]
```diff
@@ -27,19 +27,30 @@
...
+    "prepare": "husky"
   },
   "resolutions": {
     "string-width": "^4.2.0",
     "wrap-ansi": "^7.0.0",
     "postman-code-generators": "1.10.1"
   },
+  "lint-staged": {
+    "*.{js,ts,tsx,svelte}": [
+      "node --max-old-space-size=6144 node_modules/eslint/bin/eslint.js --cache --fix",
```
Comment (benoitf) [inline package.json:40]: suggestion: we don't need more memory on ai-lab compared to podman-desktop to run the linter
```suggestion
      "eslint --cache --fix",
```

**Review by benoitf — APPROVED** [review]

**Comment by maryamtahhan** [comment]
> LGTM and thank you for you patience @maryamtahhan

No problem at all. Thank you all for your help :)

**Comment by benoitf** [comment]
@maryamtahhan it seems you changed the GitHub template in the description of the PR
there is not the section to reference issues

reference issue is https://github.com/containers/podman-desktop-extension-ai-lab/issues/1633

**Pull request merged**